src/viam/sdk/registry/registry.hpp

```diff
@@ -58,6 +58,36 @@ class ResourceRegistration {
     const google::protobuf::ServiceDescriptor* service_descriptor_;
 };
 
+/// @brief subclass of ResourceRegistration with templatized defaults
+template <typename MyResourceClient,
+          typename MyResourceServer,
+          typename MyProtoService,
```

Comment:
The `MyProtoService` type parameter could probably go away if all `ResourceServer` types had a required `proto_service_type` typedef. They all inherit from that type so it would make sense:

```
class MLModelServiceServer : public ResourceServer,
                             public ::viam::service::mlmodel::v1::MLModelService::Service {
   public:
       using proto_service_type = ::viam::service::mlmodel::v1::MLModelService::Service;
   ...
};
```

Then you could just say `p->FindServiceByName(MyResourceServer::proto_service_type::service_full_name());`, and then this template would have only two types: the Client and the Server (if you get rid of the CRTP).

Of course, that would require attaching that typedef to all server classes.